agent/agent.go

```diff
@@ -21,13 +21,38 @@ import (
 	"github.com/google/gops/signal"
 )
 
-// Start starts the gops agent on a host process. Once agent started,
-// users can use the advanced gops features.
+// Agent represents an agent that enable the advanced gops features
+type Agent struct {
```

Comment:
Should we export this type? What does it mean to have multiple instances of Agent started? I'd just expect Start function to take a boolean.

`func Start(cleanUponShutdown bool) error`

instead.